src/clj/rems/css/styles.clj

```diff
@@ -272,21 +272,59 @@
                      :color "#ccc"}
                     [:&:hover {:color (util/get-theme-attribute :color4)
                                :background-color "#eee"}]]]
-   [:.alert-info
-    (s/descendant :.state-info :.phases :.phase.completed)
-    {:color (util/get-theme-attribute :info-color)
-     :background-color (util/get-theme-attribute :info-bgcolor)}]
+   
+   [:.text-primary {:color (util/get-theme-attribute :text-primary)}]
+   [:.text-secondary {:color (util/get-theme-attribute :text-secondary)}]
+   [:.text-success {:color (util/get-theme-attribute :text-success)}]
+   [:.text-danger {:color (util/get-theme-attribute :text-danger)}]
+   [:.text-warning {:color (util/get-theme-attribute :text-warning)}]
+   [:.text-info {:color (util/get-theme-attribute :text-info)}]
+   [:.text-light {:color (util/get-theme-attribute :text-light)}]
+   [:.text-dark {:color (util/get-theme-attribute :text-dark)}]
+   [:.text-muted {:color (util/get-theme-attribute :text-muted)}]
+   [:.text-white {:color (util/get-theme-attribute :text-white)}]
+
+   [:.bg-primary {:background-color (util/get-theme-attribute :bg-primary)}]
+   [:.bg-secondary {:background-color (util/get-theme-attribute :bg-secondary)}]
+   [:.bg-success {:background-color (util/get-theme-attribute :bg-success)}]
+   [:.bg-danger {:background-color (util/get-theme-attribute :bg-danger)}]
+   [:.bg-warning {:background-color (util/get-theme-attribute :bg-warning)}]
+   [:.bg-info {:background-color (util/get-theme-attribute :bg-info)}]
+   [:.bg-light {:background-color (util/get-theme-attribute :bg-light)}]
+   [:.bg-dark {:background-color (util/get-theme-attribute :bg-dark)}]
+   [:.bg-white {:background-color (util/get-theme-attribute :bg-white)}]
+
+   [:.alert-primary {:color (util/get-theme-attribute :alert-primary-color)
+                     :background-color (util/get-theme-attribute :alert-primary-bgcolor)
+                     :border-color (util/get-theme-attribute :alert-primary-bordercolor :alert-primary-color)}]
```

Comment:
If the theme attributes are not defined, this will produce CSS about which Chrome gives the error "Invalid property value":

![devtools](https://user-images.githubusercontent.com/42678/53418634-e80af300-39e0-11e9-9e99-73812c312106.png)

Reply:
I noticed that too, but does it really matter? At least Chrome seems to handle that just fine.

We could wrap everything in `styles.clj` with lets and conditionals, but I'm not sure it's worth it?

Reply:
A little function which removes all keys with nil values could do it:
```
   [:.alert-primary (remove-nils {:color (util/get-theme-attribute :alert-primary-color)
                                  :background-color (util/get-theme-attribute :alert-primary-bgcolor)
                                  :border-color (util/get-theme-attribute :alert-primary-bordercolor :alert-primary-color)})]
```

Reply:
Put a default for everything or don't put nil?

Reply:
Esko has a good point, I'll just remove nil values.

Reply:
```
(when-let [style (remove-nil-vals {:color (util/get-theme-attribute :alert-dark-color)
                                      :background-color (util/get-theme-attribute :alert-dark-bgcolor)
                                      :border-color (util/get-theme-attribute :alert-dark-bordercolor :alert-dark-color)})]
     [:.alert-dark style])
```
I wrapped the new classes like this. Seems to work.

I actually implemented  a recursive function that walks through the whole list of styles and removes nil values. I got it working (I think), but in the end I decided that it's too complicated, and thus prone to bugs. While this approach is quite verbose, it's at least clear what is being done.